pkgs/by-name/li/linuxhw-edid-fetcher/linuxhw-edid-fetch.sh

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env bash
+set -eEuo pipefail
+test -z "${DEBUG:-}" || set -x
+# based on instructions provided in https://github.com/linuxhw/EDID/blob/98bc7d6e2c0eaad61346a8bf877b562fee16efc3/README.md
+
+usage() {
+  cat <<'EOF'
+Example invocations:
+  linuxhw-edid-fetcher /path/to/github.com/linuxhw/EDID PG278Q 2014 >edid.bin
+  edid-decode <edid.bin
+  parse-edid <edid.bin
+  cat edid.bin >/sys/kernel/debug/dri/0/DP-1/edid_override
+EOF
+}
+
+print() {
+  # shellcheck disable=SC2059
+  printf "${1}\n" "${@:2}"
+}
+
+log() {
+  print "${@}" >&2
+}
+
+find_displays() {
+  local script=(
+    "BEGIN { IGNORECASE=1 }"
+    "/${1}/"
+  )
+
+  for pattern in "${@:2}"; do
+    script+=('&&' "/${pattern}/")
+  done
+  cat "${repo}"/{Analog,Digital}Display.md | awk "${script[*]}"
+}
+
+to_edid() {
+  if ! test -e "$1"; then
+    log "$1 does not exist"
+    return 1
+  fi
+
+  # https://github.com/linuxhw/EDID/blob/228fea5d89782402dd7f84a459df7f5248573b10/README.md#L42-L42
+  grep -E '^([a-f0-9]{32}|[a-f0-9 ]{47})$' <"$1" | tr -d '[:space:]' | xxd -r -p
+}
+
+extract_path() {
+  awk '{ gsub(/^.+]\(</, ""); gsub(/>).+/, ""); print }'
+}
+
+main() {
+  if [[ $# == 0 ]] ; then
+    usage
+    exit 1
+  fi
+
+  repo="${repo:-"$PWD"}"
+
+  if [[ -d "$1" ]]; then
+    repo="$1"
+    shift
+  fi
```

Comment:
I'd prefer if this assumed the synopsis on line 9, rather than allowing a default:
```suggestion
  repo="$1"
  shift
```

The checks on line 65 will take care of verifying that `$1` points to a directory.

Reply:
I actually intend it to default to `${src}` for a package, but also want it to make sense on it's own (outside Nix).

Reply:
Having two separate synopses, one which has to be detected by a hack (checking if the first argument is a directory) isn't ideal though. What if you call `linuxhw-edid-fetcher PG278Q 2014 >edid.bin`, and there happens to be a `PG278Q` directory? It would do the wrong thing. Better to do something explicit, like

```bash
if [[ $# -eq 3 ]]; then
    do the directory thing
elif [[ $# -eq 2 ]]; then
    do the non-directory thing
else
    usage >&2
    exit 2
fi
```

Reply:
counting arguments definitely does not make sense in here as the search can have arbitrary number of arguments depending on the specific device. I actually expected `PG278Q` to be enough for mine, but it turns out there are over 10 flavors right now (used to be ~4 when I originally written the script)

Reply:
reworked the thing a bit not to use the argument at all and giving example on overriding the directory using `repo=/path/to/EDID`